src/MongoDB.Driver.Core/Core/Authentication/MongoAWSAuthenticator.cs

```diff
@@ -0,0 +1,539 @@
+
+            if (relativeUri == null)
+            {
+                return null;
+            }
+
+            var response = AwsHttpClientHelper.GetECSResponseAsync(relativeUri).GetAwaiter().GetResult();
+            var parsedResponse = BsonDocument.Parse(response);
+            var username = parsedResponse.GetValue("AccessKeyId", null)?.AsString;
+            var password = parsedResponse.GetValue("SecretAccessKey", null)?.AsString;
+            var sessionToken = parsedResponse.GetValue("Token", null)?.AsString;
+
+            var awsCredentials = new AwsCredentials
+            {
+                Username = username,
+                Password = ToSecureString(password),
+                SessionToken = sessionToken
+            };
+
+            ValidateAwsCredentials(awsCredentials);
+
+            if (awsCredentials.Username == null)
+            {
+                return null;
+            }
+
+            return awsCredentials;
+        }
+
+        private static AwsCredentials CreateAwsCredentialsFromEc2Response()
+        {
+            var response = AwsHttpClientHelper.GetEC2ResponseAsync().GetAwaiter().GetResult();
+            var parsedResponse = BsonDocument.Parse(response);
+            var username = parsedResponse.GetValue("AccessKeyId", null)?.AsString;
+            var password = parsedResponse.GetValue("SecretAccessKey", null)?.AsString;
+            var sessionToken = parsedResponse.GetValue("Token", null)?.AsString;
+
+            var awsCredentials = new AwsCredentials
+            {
+                Username = username,
+                Password = ToSecureString(password),
+                SessionToken = sessionToken
+            };
+
+            ValidateAwsCredentials(awsCredentials);
+
+            if (awsCredentials.Username == null)
+            {
+                return null;
+            }
+
+            return awsCredentials;
+        }
+
+        private static string ExtractSessionTokenFromMechanismProperties(IEnumerable<KeyValuePair<string, string>> properties)
+        {
+            if (properties == null)
+            {
+                return null;
+            }
+
+            foreach (var pair in properties)
+            {
+                if (pair.Key.ToUpperInvariant() == "AWS_SESSION_TOKEN")
+                {
+                    return pair.Value;
+                }
+            }
+
+            return null;
+        }
+
+        private static SecureString ToSecureString(string str)
+        {
+            if (str == null)
+            {
+                return null;
+            }
+
+            var secureString = new SecureString();
+            foreach (var c in str)
+            {
+                secureString.AppendChar(c);
+            }
+            secureString.MakeReadOnly();
+
+            return secureString;
+        }
+
+        private static void ValidateAwsCredentials(AwsCredentials awsCredentials)
```

Comment:
Remove this method, it is no longer needed.

A similar validation is now done in the `AwsCredentials` constructor.